### PR TITLE
Architecture change for non-prefix commands

### PR DESCRIPTION
**Pull request by charlotte-zhuang**
1. Moved handlers to handlers folder
2. created MessageHandler to handle all messages
3. changed CommandHandler to only handle prefixed commands
3. created nonPrefixHanlder to handle non-prefix commands
4. created nonPrefixRegexCreator to match all non-prefix commands
5. created nonPrefixCommands directory and Commands class for non-prefix commands
6. created Thread and ThreadManager prefixed command boilerplates
7. created ThreadMessage non-prefixed command boilerplate

### CODE REVIEW

File: src/handlers/MessageHandler.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const handlersPath = '.';
+const utilPath = '../util';
+
+const { CommandHandler } = require(handlersPath + '/CommandHandler');
+const { NonPrefixHandler } = require(handlersPath + '/nonPrefixHandler');
```
Comment (jerrylee17) [inline src/handlers/MessageHandler.js:5]: Capitalize `NonPrefixHandler`

File: src/nonPrefixCommands/threads/threadMessage.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+const Command = require('../Command');
+
+module.exports = new Command({
+  name: 'no-prefix threadmessage',
+  regex: new RegExp(/^\|(\d{1,18})\|\s+/),
```
Comment (jerrylee17) [inline src/nonPrefixCommands/threads/threadMessage.js:6]: Personal preference - I like having the space before/after the thread ID: `/^\|\s*(\d{1,18})\s*\|\s+/`

**Review by jerrylee17 — CHANGES_REQUESTED** [review]
image
Capitalize the thingy in messageHandler.

**Review by jerrylee17 — APPROVED** [review]
Nice job!

**Comment by jerrylee17** [comment]
@charlotte-zhuang you can probably merge this in

**Comment by charlotte-zhuang** [comment]
Sorry, I didn't realize I needed to merge my PR.

**Pull request merged**